crowbar_framework/lib/crowbar/upgrade_status.rb

```diff
@@ -420,6 +416,26 @@ def next_step_to_execute
       upgrade_steps[i + 1]
     end
 
+    def upgrade_finished_recently?
+      return false unless File.exist?(finished_file)
+      # here "recently" means within last 24hrs
+      (Time.now.to_i - File.read(finished_file).to_i) < 24 * 60 * 60
```

Comment:
So we know that the file exists, but is it readable and does it have the right content? Should we add some exception handling?

Reply:
in the end the question is: what should we return from this function in case of any problem with the file? can we assume that "upgrade didn't finish recently" in those cases?

Reply:
That's the right question indeed, but I'm not sure about the answer.
If the upgrade has failed for some reason, we should not even get to this place, and `running` file should still exist.
It could be broken when writing (let's say, because of full disk), but that does not mean upgrade itself is unfinished.
Or user could have modified it.

Reply:
if the file was not corrupted or modified, the mtime should more or less match the contents.

Reply:
Make sense.